parley/src/layout/editor.rs

```diff
@@ -655,12 +707,27 @@ where
     }
 
     /// Select the physical line at the point.
+    ///
+    /// Note that this metehod determines line breaks for any reason, including due to word wrapping.
+    /// To select the text between explicit newlines, use [`select_paragraph_at_point`](Self::select_paragraph_at_point).
+    /// In most text editing cases, this is the preferred behaviour.
+    // TODO: Rename to `select_soft_line_at_point`? Very few users should use this method.
```

Comment:
I think that's a fair change, maybe there should still be `select_line_at_point` for explicit lines; this is a desirable behavior in code editors.

Reply:
Oh, sorry I misread the code, I thought you were checking for start of text, end of text, and _two consecutive hard line breaks_. A single hard line break is just `select_line_at_point`.